### ISSUE TITLE
Question: Does nexline have an internal buffer?

### ISSUE THREAD

**Issue by sbrl** [sharpart555/nexline#36]
For my next trick,  I'm planning to share `process.stdin` of a master control process across multiple worker processes. From there, I'm want to read from said file descriptor line-by-line from the multiple worker processes.

You can probably see where this is going, but I'd like to know whether `nexline` has an internal buffer. If so, this could complicate matters considerably, as the buffer will be local to a single worker process as opposed to shared across all workers.

Many thanks in advance :-)

**Comment by sharpart555** [comment]
Hi again :) 
The answer is **yes**, `nexline` has internal buffer and cannot read a line without it.
* `nexline` supports multiple and custom line separators. So `nexline` needs to read data until all line separator is checked.
* `nexline` uses `fs.read()` for file descriptor and it does not support "reading until specific string"
* If `nexline` reads too small data at a time, performance will drop a lot.

**Comment by sbrl** [comment]
Hello again!

Ah, I see - I suspected that. Thanks for the info. I ended up [implementing a workaround](https://starbeamrainbowlabs.com/blog/article.php?article=posts/416-multiprocess-line-based-read-nodejs.html) that reads 1 character at a time with  `fs.read()`, though I have yet to test it for performance - as I haven't finished the actual implementation yet.

If performance is an issue, I'm probably going to have to re-evaluate my approach to the issue. My scenario is that I have a number of compressed files that 1 large JSON object per line. I've tried reading it all in a master thread and sending it to the workers, but that's a bottleneck because the master is tied up copying the data to the workers, and the workers never get enough work to do.

So if my workaround doesn't work, then I'll probably end up implementing something like a Bash script that first starts the Node.js process, waits for it to start listening on a TCP socket, and then proceeds to read the file line-by-line, making lots of TCP connections to the Node.js cluster of process. I've no idea how I'd ensure that  they all get written out to the output after processing in the right order though.....

But that's an issue for me to solve for my PhD.

Thanks again!
